### PR TITLE
crimson/osd/ops_executer: LIST_SNAPS only on CEPH_SNAPDIR

### PR DESCRIPTION
**Pull request by guojidan**
fixes: https://tracker.ceph.com/issues/65663

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by Matan-B — APPROVED** [review]
crimson/osd/ops_executer.cc: LIST_SNAPS only on CEPH_SNAPDIR

**Comment by guojidan** [comment]
jenkins test make check

**Comment by guojidan** [comment]
jenkins test make check arm64

**Comment by guojidan** [comment]
jenkins test windows

**Comment by Matan-B** [comment]
> crimson/osd/ops_executer: LIST_SNAPS only on CEPH_SNAPDIR

@guojidan, Can you please rename the commit to: `crimson/osd/ops_executer: LIST_SNAPS only on CEPH_SNAPDIR`?

**Comment by guojidan** [comment]
> > crimson/osd/ops_executer: LIST_SNAPS only on CEPH_SNAPDIR
> 
> @guojidan, Can you please rename the commit to: `crimson/osd/ops_executer: LIST_SNAPS only on CEPH_SNAPDIR`?

OK, thanks

**Comment by guojidan** [comment]
jenkins test windows

File: src/crimson/osd/ops_executer.cc [hunk 1]
```diff
@@ -489,6 +489,11 @@ OpsExecuter::list_snaps_iertr::future<> OpsExecuter::do_list_snaps(
   const ObjectState& os,
   const SnapSet& ss)
 {
+  if (msg->get_snapid() != CEPH_SNAPDIR) {
+    logger().error("LIST_SNAPS with incorrect context");
```
Comment (athanatos) [inline src/crimson/osd/ops_executer.cc:493]: No need to log at error here -- it's not actually a problem for the OSD.

Comment (guojidan) [reply src/crimson/osd/ops_executer.cc:493]: yes, should be debug level similar to classical osd

**Comment by Matan-B** [comment]
https://shaman.ceph.com/builds/ceph/wip-matanb-crimson-testing-june-9-E/

**Comment by Matan-B** [comment]
jenkins test windows

**Comment by Matan-B** [comment]
Merging based on (21/21 passed):
https://pulpito.ceph.com/matan-2024-06-09_12:08:27-crimson-rados-wip-matanb-crimson-testing-june-9-E-distro-crimson-smithi/
Rereun dead job:
https://pulpito.ceph.com/matan-2024-06-09_13:09:45-crimson-rados-wip-matanb-crimson-testing-june-9-E-distro-crimson-smithi/

**Pull request merged**